lib/Cake/Network/CakeSocket.php

```diff
@@ -406,5 +406,14 @@ public function enableCrypto($type, $clientOrServer = 'client', $enable = true)
 		throw new SocketException($errorMessage);
 	}
 
+/**
+ * Accept Self-signed certificate on current stream socket
+ *
+ * @return bool True on success
+ * @see stream_context_set_option
```

Comment:
``@link http://php.net/manual/en/context.ssl.php About the 'allow_self_signed' option.`` might be better or good in addition.